framework/src/auxkernels/MaterialStdVectorAux.C

```diff
@@ -34,5 +37,8 @@ MaterialStdVectorAux::~MaterialStdVectorAux()
 Real
 MaterialStdVectorAux::getRealValue()
 {
+  mooseAssert(_special_qp < _q_point.size(), "MaterialStdVectorAux.  special_qp specified as " << _special_qp << " but there are only " << _q_point.size() << " quadpoints in the element");
```

Comment:
If only there was another floating point value that represented the fact that what we return is _not a number_...
Another option would be to return a value from std::limits, but unlike NaN this seriously messes with the Paraview autoscaling. NaN just seems like the most elegant solution here.